sparkly/utils.py

```diff
@@ -86,3 +91,162 @@ def kafka_get_topics_offsets(host, topic, port=9092):
                         end_offset.offsets[0]))
 
     return offsets
+
+
+def parse_schema(schema):
+    """Converts string to Spark schema definition.
+
+    Usages:
+        >>> parse('struct[a:struct[a:string]]').simpleString()
+        'struct<a:struct<a:string>>'
+
+    Args:
+        schema (str): Schema definition as string.
+
+    Returns:
+        StructType
+
+    Raises:
+        UnsupportedDataType: In case of unsupported data type.
+    """
+    return _generate_structure_type(_parse_schema(schema))
+
+
+def _generate_structure_type(fields_and_types):
+    """Generate a StructType from the dict of fields & types.
+
+    Schema definition supports basic types: string, integer, long, float, boolean.
+    And complex types in any combinations: dict, struct, list.
+
+    Usages:
+        >>> _generate_structure_type({'field_a': 'long'}).simpleString()
+        'struct<field_a:bigint>'
+        >>> _generate_structure_type({'field_a': 'dict[string,long]'}).simpleString()
+        'struct<field_a:map<string,bigint>>'
+        >>> _generate_structure_type({'field_a': 'loooong'})
+        Traceback (most recent call last):
+        ...
+        sparkly.exceptions.UnsupportedDataType: Unsupported type field_a for field loooong
+
+    Args:
+        fields_and_types (dict[str, str]): Field - type associations.
+            Possible types are string, integer, long, float, boolean, dict, struct.
+
+    Returns:
+        StructType
+
+    Raises:
+        UnsupportedDataType: In case of unsupported data type.
+    """
+    struct = StructType()
+    for field_name, field_type in fields_and_types.items():
+        try:
+            struct.add(field_name, _process_type(field_type))
+        except UnsupportedDataType:
+            message = 'Unsupported type {} for field {}'.format(field_type, field_name)
+            raise UnsupportedDataType(message)
+
+    return struct
+
+
+def _parse_schema(schema):
+    """Converts schema string to dict: field_name -> type definition string.
+
+    Note:
+        We need an OrderedDict here because ordering matters in Dataframe definition
+        when I try to apply it to RDD, this is what we do when read kafka RDD.
```

Comment:
this line might read better as:

when applied to a RDD, like when we read from Kafka RDDs.